core/src/main/scala/scheduler/KubernetesShell.scala

```diff
@@ -70,7 +70,15 @@ final class KubernetesShell(
       }
     }
 
-    val template = blueprint.fold(fallback)(IO.pure)
+    // NOTE: by this point in the system, we know we're dealing with
+    // a hydrated blueprint (i.e. passes manifest validation and exists
+    // in the database) so we simply take the supplied plan and extract
+    // the `Blueprint`, and `Template` in turn.
+    val template = plan.environment.blueprint
+      .flatMap(_.right.toOption)
```

Comment:
This can have hidden/strange behavior if we have an internal bug and provide an unsaturated (hydrated?) blueprint and silently fallback to a default - see type parameter trick above.